### PR TITLE
Use helix reporting

### PR DESCRIPTION
**Pull request by ChadNedzlek**
Use the Helix test reporting feature that allows for test retry configuration/reporting.

### CODE REVIEW

**Review by monstro805 — APPROVED** [review]

**Review by monstro805 — APPROVED** [review]

**Review by garath — APPROVED** [review]

File: src/Microsoft.DotNet.Helix/Sdk/tools/.env [hunk 1]
```diff
@@ -0,0 +1 @@
+PYTHONPATH=C:\Program Files (x86)\Microsoft Visual Studio\Shared\Python37_64\lib;D:\vsgit\machines\resources\helix-scripts
```
Comment (garath) [inline src/Microsoft.DotNet.Helix/Sdk/tools/.env:1]: Is this file meant to be checked in?

File: src/Microsoft.DotNet.Helix/Sdk/tools/azure-pipelines/reporter/formats/result_format.py [hunk 2]
```diff
@@ -10,15 +10,13 @@ def __init__(self):
         pass
 
     @abstractproperty
-    def name(self):
+    def name(self) -> str:
```
Comment (garath) [inline src/Microsoft.DotNet.Helix/Sdk/tools/azure-pipelines/reporter/formats/result_format.py:13]: Does this typing work on the versions of python in use?

Comment (ChadNedzlek) [reply src/Microsoft.DotNet.Helix/Sdk/tools/azure-pipelines/reporter/formats/result_format.py:13]: Helix only works on 3 or above, so typings like this should be fine.  My understanding is that it's just the `var:type = value` ones that don't work in 3.5 or lower.

File: src/Microsoft.DotNet.Helix/Sdk/tools/.env (hunk 1 above)
Comment (ChadNedzlek) [reply src/Microsoft.DotNet.Helix/Sdk/tools/.env:1]: No, not it's not. Let me get a .gitignore update for it...

**Review by AlitzelMendez — APPROVED** [review]

**Comment by MattGal** [comment]
It would be very beneficial with this reporting change to add support for a docker-container helix work item, so we find out whether this works there before going to production.

**Comment by missymessa** [comment]
> It would be very beneficial with this reporting change to add support for a docker-container helix work item (just a row in UnitTests.proj), so we find out whether this works there before going to production.

@MattGal would you mind writing an issue for this work? Since Chad is OOF for another week, we are likely going to merge this PR in as-is before he gets back if it's successful.

**Comment by MattGal** [comment]
> > It would be very beneficial with this reporting change to add support for a docker-container helix work item (just a row in UnitTests.proj), so we find out whether this works there before going to production.
> 
> @MattGal would you mind writing an issue for this work? Since Chad is OOF for another week, we are likely going to merge this PR in as-is before he gets back if it's successful.

It's just one line to the project file to prove that this scenario works, I really think either adding it or manually exercising this with it should be part of this PR before it merges.  If you get everything else working but that's still a problem, I can help.

**Comment by missymessa** [comment]
> > > It would be very beneficial with this reporting change to add support for a docker-container helix work item (just a row in UnitTests.proj), so we find out whether this works there before going to production.
> > 
> > 
> > @MattGal would you mind writing an issue for this work? Since Chad is OOF for another week, we are likely going to merge this PR in as-is before he gets back if it's successful.
> 
> It's just one line to the project file to prove that this scenario works, I really think either adding it or some proof of manually exercising this with it should be part of this PR before it merges. If you get everything else working but that's still a problem, I can help.

Would you mind contributing that change to this PR?

**Comment by MattGal** [comment]
> Would you mind contributing that change to this PR?

Sure, but let's wait til the normal path is green in the current run to make sure we're not mixing and matching different problems.

**Review by MattGal — APPROVED** [review]
I will open a PR immediately after this merges to add the docker testing I commented about.

**Comment by missymessa** [comment]
LGTM shipit

**Pull request merged**